unity-renderer/Assets/DCLServices/MapRendererV2/MapRenderer.cs

```diff
@@ -156,37 +159,47 @@ public void SetSharedLayer(MapLayer mask, bool active)
             }
         }
 
-        private void EnableLayers(MapLayer mask)
+        private void EnableLayers(IMapActivityOwner owner, MapLayer mask)
         {
             foreach (MapLayer mapLayer in ALL_LAYERS)
             {
                 if (!EnumUtils.HasFlag(mask, mapLayer) || !layers.TryGetValue(mapLayer, out MapLayerStatus mapLayerStatus)) continue;
 
-                if (mapLayerStatus.ActivityOwners == 0 && mapLayerStatus.sharedActive != false)
+                if (owner.LayersParameters.TryGetValue(mapLayer, out IMapLayerParameter parameter))
+                    mapLayerStatus.MapLayerController.SetParameter(parameter);
+
+                if (mapLayerStatus.ActivityOwners.Count == 0 && mapLayerStatus.SharedActive != false)
                 {
                     // Cancel deactivation flow
                     ResetCancellationSource(mapLayerStatus);
                     mapLayerStatus.MapLayerController.Enable(mapLayerStatus.CTS.Token).SuppressCancellationThrow().Forget();
                 }
 
-                mapLayerStatus.ActivityOwners++;
+                mapLayerStatus.ActivityOwners.Push(owner);
             }
         }
 
-        private void DisableLayers(MapLayer mask)
+        private void DisableLayers(IMapActivityOwner owner, MapLayer mask)
         {
             foreach (MapLayer mapLayer in ALL_LAYERS)
             {
                 if (!EnumUtils.HasFlag(mask, mapLayer) || !layers.TryGetValue(mapLayer, out MapLayerStatus mapLayerStatus)) continue;
 
-                mapLayerStatus.ActivityOwners = Math.Max(mapLayerStatus.ActivityOwners - 1, 0);
+                mapLayerStatus.ActivityOwners.Pop();
```

Comment:
You should respect `owner` parameter as otherwise the flow can be broken if the disabled owner is not the last one that was pushed. Thus, I suppose `Stack` is not the right choice: a simple `List` will be better

Reply:
changed to List and checked if it contains `owner` before removing it